src/filter.c

```diff
@@ -79,9 +88,92 @@ static char* readln(gzFile* f) {
     return line;
 }
 
+
 char* (*read_func)(gzFile*) = readln;
 
 
+static char* get_tile_id(char* fastq_header) {
+    char* hdr = malloc(sizeof (char) * strlen(fastq_header));
+    strcpy(hdr, fastq_header);  // strtok modifies the string passed to it, so use a copy
+    
+    char* field;
+    field = strtok(hdr, ":");
+    int i;
+    for (i=0; i<4; i++) {
+        field = strtok(NULL, ":");  // walk along the header 4 times to the tile ID
+    }
+    free(hdr);
+    return field;
+}
+
+
+
+static bool std_check_read(char* r1_header, char* r1_seq, char* r1_strand, char* r1_qual, char* r2_header, char* r2_seq, char* r2_strand, char* r2_qual) {
+    if ((strlen(r1_seq) > threshold) && (strlen(r2_seq) > threshold)) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+
+static bool tile_check_read(char* r1_header, char* r1_seq, char* r1_strand, char* r1_qual, char* r2_header, char* r2_seq, char* r2_strand, char* r2_qual) {
+    int result = std_check_read(r1_header, r1_seq, r1_strand, r1_qual, r2_header, r2_seq, r2_strand, r2_qual);
+    if (result == false) {
+        return false;
+    }
+    
+    char* tile_id = get_tile_id(r1_header);
+    int i = 0;
+    
+    char* comp = remove_tiles[i];
+    while (comp != NULL) {  // check for null terminator at end of remove_tiles
+        if (strcmp(comp, tile_id) == 0) {
+            return false;
+        }
+        i++;
+        comp = remove_tiles[i];
+    }
+    
+    return true;
+}
+
+
+bool (*check_func)(char*, char*, char*, char*, char*, char*, char*, char*) = std_check_read;
+
+
+static void std_include(char* header, char* seq, char* strand, char* qual, FILE* outfile) {
+    fputs(header, outfile);
+    fputs(seq, outfile);
+    fputs(strand, outfile);
+    fputs(qual, outfile);
+}
+
+
+static void _trim_include(char* header, char* seq, char* strand, char* qual, FILE* outfile, int trim_len) {
+    if (strlen(seq) > trim_len + 1) {  // add 1 here to compensate for \n at end of line...
+        seq[trim_len] = '\n';  // ...but 0-indexing means we don't need to add 1 here
+        seq[trim_len + 1] = '\0';
+        qual[trim_len + 1] = '\n';  // qual line has a # at the start, so add 1
```

Comment:
I don't think this is the case.
Seq and qal line have exactly the same length.
At the beginning of files there might be a lot of reads that have bad quality in the first cycle `#` is qual 2 but there should not be any different treatment between seq and qual lines